internal/oci/runtime_oci.go

```diff
@@ -898,73 +898,93 @@ func (r *runtimeOCI) AttachContainer(c *Container, inputStream io.Reader, output
 	return nil
 }
 
-// PortForwardContainer forwards the specified port provides statistics of a container.
-func (r *runtimeOCI) PortForwardContainer(c *Container, port int32, stream io.ReadWriter) error {
-	emptyStreamOnError := true
-	defer func() {
-		if emptyStreamOnError {
-			go func() {
-				if _, copyError := pools.Copy(ioutil.Discard, stream); copyError != nil {
-					logrus.Errorf("error closing port forward stream after other error: %v", copyError)
-				}
-			}()
-		}
-	}()
+// PortForwardContainer forwards the specified port into the provided container.
+func (r *runtimeOCI) PortForwardContainer(
+	c *Container, port int32, stream io.ReadWriteCloser,
+) error {
 	containerPid := c.State().Pid
-	socatPath, lookupErr := exec.LookPath("socat")
-	if lookupErr != nil {
-		return fmt.Errorf("unable to do port forwarding: socat not found")
-	}
+	netNsPath := fmt.Sprintf("/proc/%d/ns/net", containerPid)
```

Comment:
I'm not the hugest fan of using /proc here... we have the netNsPath saved in the sandbox, can you do something similar to https://github.com/cri-o/cri-o/pull/3474/commits/546e9817d85eeb27211706cb663e69e9d0348b10 instead?

Reply:
this could also be a follow up if you'd prefer

Reply:
Yep, fixed as suggested. :+1: